src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -256,6 +263,34 @@ private static ExecutorService newExecutorService() {
             }
         }
 
+        @Override
+        public void onFailure(StepContext context, Throwable t) {
+            super.onFailure(context, t);
```

Comment:
I think it would make more sense to call this at the end of the method rather than the start so that the exception is updated before `context.onFailure(t)` is called in the superclass. I guess it's fine since this code occupies the CPS VM thread so other steps will not run until everything returns, but it seems confusing.

Reply:
Oh, I just saw https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/144#discussion_r643216989. Like I mentioned above, I don't think it matter because of the CPS VM thread, but I'm not 100% sure. If there is an issue, I guess it would be some kind of race condition where an enclosing step sees the failure before the exception gets updated.

Reply:
I moved it to the end as you suggested.